Dockerfile

```diff
@@ -0,0 +1,47 @@
+FROM node:18-bullseye-slim AS base
+
+# This doesn invalidate layer cache if the result from turbo prune stays the same
+# this step generates a trimmed down version of the package list for installer
+FROM base AS builder
+ARG APP
```

Comment:
You can declare the `APP` ARG on line 2 and then I think it's valid for the whole file. ARG behaves in such a way that cache misses will only occur when the ARG is used with a value it hasn't been used before.

Reply:
@joonatank Actually, this doesn't seem to be the case, see: https://docs.docker.com/engine/reference/builder/#understand-how-arg-and-from-interact

Somehow backend is able to build using this method locally, but I'm getting errors in Azure Pipelines for missing args.

Reply:
This might be a [BuildKit](https://docs.docker.com/build/buildkit/) vs OpenShift thing.